File: app/src/main/java/com/martinrevert/latorrentola/database/MovieDao.java
package com.martinrevert.latorrentola.database;

import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.OnConflictStrategy;
import androidx.room.Query;

import com.martinrevert.latorrentola.model.YTS.Movie;

import java.util.List;

import io.reactivex.Flowable;
import io.reactivex.Single;


/**
 * Created by martin on 07/12/17.
 */
@Dao
public interface MovieDao {

    @Query("SELECT * FROM movie")
    Flowable<List<Movie>> getAll();

    @Query("SELECT * FROM movie WHERE id LIKE :id")
    Single<Movie> getMovie(Integer id);

    @Delete
    void delete(Movie movie);

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    void insertMovie(Movie movie);
}

File: app/src/main/java/com/martinrevert/latorrentola/database/MovieDao.kt
package com.martinrevert.latorrentola.database

import androidx.room.Dao
import androidx.room.Delete
import androidx.room.Insert
import androidx.room.OnConflictStrategy
import androidx.room.Query
import com.martinrevert.latorrentola.model.YTS.Movie
import kotlinx.coroutines.flow.Flow

@Dao
interface MovieDao {

    @Query("SELECT * FROM movies")
    fun getAll(): Flow<List<Movie>>

    @Query("SELECT * FROM movies WHERE id = :id")
    suspend fun getMovie(id: Int): Movie?

    @Delete
    suspend fun delete(movie: Movie)

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    suspend fun insertMovie(movie: Movie)
}
